### PR TITLE
Add new nested resource Microsoft.AAD/DomainServices/ReplicaSets

### PR DESCRIPTION
**Pull request by jihochang**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by jihochang** [comment]
@majastrz - Adding for review

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json [hunk 1]
```diff
@@ -311,6 +311,242 @@
...
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AAD/domainServices/{domainServiceName}/replicaSets": {
+      "get": {
+        "tags": [
+          "ReplicaSets"
+        ],
+        "operationId": "ReplicaSets_ListByResourceGroup",
+        "summary": "List ReplicaSets in Resource Group (GET Resources)",
```
Comment (kpajdzik) [inline specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json]: What's the purpose of `(GET Resources)` part?

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json [hunk 2]
```diff
@@ -311,6 +311,242 @@
...
+        ],
+        "responses": {
+          "202": {
+            "description": "HTTP 202 (Accepted) if the operation was successfully started and will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/ReplicaSet"
+            }
+          },
+          "204": {
+            "description": "HTTP 204 (Not Content) should be used if the resource does not exist and the request is well formed."
```
Comment (majastrz) [inline specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json]: >HTTP 204 (Not Content) should be used if the resource does not exist and the request is well formed. [](start = 28, length = 100)

I think this description should say when the service returns 204. Perhaps something like: "HTTP 204 (No Content) if the resource does not exist."

**Review by kpajdzik — APPROVED** [review]

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json (hunk 1 above)
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json]: I can remove that.

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json (hunk 2 above)
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json]: changed.

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json [hunk 3]
```diff
@@ -473,6 +741,68 @@
         }
```
Comment (majastrz) [inline specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json]: Adding new properties to an existing resource in the same API version is considered a breaking change. Can you introduce a new API version?

**Review by majastrz — CHANGES_REQUESTED** [review]
I have a concern about introducing new properties on existing resource types in a stable API version. Can we address that?

File: specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json (hunk 3 above)
Comment (jihochang) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json]: We are not making any changes to the existing domain service root resource and just adding a new nested resource. Do we still need to rev to a new API version?

Comment (majastrz) [reply specification/domainservices/resource-manager/Microsoft.AAD/stable/2017-06-01/domainservices.json]: It looks like a diff bug. I was seeing an addition of 3 properties to the existing resource schema. Compared manually against what is committed. There's no need to introduce a new API version.

**Review by majastrz — APPROVED** [review]

**Comment by kpajdzik** [comment]
@jihochang, do you want me to merge this PR?

**Comment by jihochang** [comment]
> @jihochang, do you want me to merge this PR?
Yes, could you please merge this PR. 
Thank you for reviewing!

**Pull request merged**